Listenbrainz/UI/Screens/ContentView.swift

```diff
@@ -1,52 +1,67 @@
 // ContentView.swift
 import SwiftUI
+import Combine
 
 struct ContentView: View {
     let screenWidth = UIScreen.main.bounds.size.width
     let screenHeight = UIScreen.main.bounds.size.height
     @EnvironmentObject var homeViewModel: HomeViewModel
     @EnvironmentObject var feedViewModel: FeedViewModel
     @StateObject var yimViewModel = YIMViewModel(repository: YIMRepositoryImpl())
-    
+    @State private var keyboardHeight: CGFloat = 0
+
     init() {
         UITabBar.appearance().layer.borderColor = UIColor.clear.cgColor
         UITabBar.appearance().clipsToBounds = true
     }
-    
+
     var body: some View {
         ZStack {
             TabView {
                 FeedView()
                     .environmentObject(feedViewModel)
-                    .frame(width: screenWidth, height: screenHeight, alignment: .center)
+                    .frame(width: screenWidth, height: screenHeight - keyboardHeight, alignment: .center)
                     .tabItem {
                         Label("Feed", systemImage: "bolt")
                     }
-                
+
                 ExploreView(viewModel: yimViewModel)
                     .edgesIgnoringSafeArea(.all)
-                    .frame(width: screenWidth, height: screenHeight, alignment: .center)
+                    .frame(width: screenWidth, height: screenHeight - keyboardHeight, alignment: .center)
                     .tabItem {
                         Label("Explore", systemImage: "safari")
                     }
-                
+
                 PlayerView()
                     .edgesIgnoringSafeArea(.all)
-                    .frame(width: screenWidth, height: screenHeight, alignment: .center)
+                    .frame(width: screenWidth, height: screenHeight - keyboardHeight, alignment: .center)
                     .tabItem {
                         Label("Player", systemImage: "headphones")
                     }
-                
+
                 ListensView()
                     .environmentObject(homeViewModel)
-                    .frame(width: screenWidth, height: screenHeight, alignment: .center)
+                    .frame(width: screenWidth, height: screenHeight - keyboardHeight, alignment: .center)
                     .tabItem {
                         Label("Listens", systemImage: "person.wave.2")
                     }
             }
             .accentColor(Color.gray)
         }
+        .onAppear {
```

Comment:
This is a really bad solution. We cannot be doing this everywhere. The way to solve this problem is probably by adding ignoreSafeArea or something to your view.

Reply:
.ignoresSafeArea(.keyboard) - I have tried using this modifier, but this was not working